SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -495,3 +435,101 @@ void CSpeciesFlameletSolver::BC_ConjugateHeat_Interface(CGeometry* geometry, CSo
 }
+
+unsigned long CSpeciesFlameletSolver::SetScalarSources(CConfig* config, CFluidModel* fluid_model_local,
+                                                       unsigned long iPoint, vector<su2double>& scalars) {
+  /*--- Compute total source terms from the production and consumption. ---*/
+
+  vector<su2double> table_sources(config->GetNControlVars() + 2*config->GetNUserScalars());
+  unsigned long misses = fluid_model_local->EvaluateDataSet(scalars, FLAMELET_LOOKUP_OPS::SOURCES, table_sources);
+  nodes->SetTableMisses(iPoint, misses);
+
+  /*--- The source term for progress variable is always positive, we clip from below to makes sure. --- */
+
+  vector<su2double> source_scalar(config->GetNScalars());
+  for (auto iCV=0u; iCV<config->GetNControlVars(); iCV++)
+    source_scalar[iCV] = table_sources[iCV];
+  source_scalar[I_PROGVAR] = fmax(EPS, source_scalar[I_PROGVAR]);
+
+  /*--- Source term for the auxiliary species transport equations. ---*/
+  for (size_t i_aux = 0; i_aux < config->GetNUserScalars(); i_aux++) {
+    /*--- The source term for the auxiliary equations consists of a production term and a consumption term:
+          S_TOT = S_PROD + S_CONS * Y ---*/
+    su2double y_aux = scalars[config->GetNControlVars() + i_aux];
+    su2double source_prod = table_sources[config->GetNControlVars() + 2 * i_aux];
+    su2double source_cons = table_sources[config->GetNControlVars() + 2 * i_aux + 1];
+    source_scalar[config->GetNControlVars() + i_aux] = source_prod + source_cons * y_aux;
+  }
+  for (auto i_scalar = 0u; i_scalar < nVar; i_scalar++)
+    nodes->SetScalarSource(iPoint, i_scalar, source_scalar[i_scalar]);
+  return misses;
+}
+
+unsigned long CSpeciesFlameletSolver::SetScalarLookUps(CConfig* config, CFluidModel* fluid_model_local,
+                                                       unsigned long iPoint, vector<su2double>& scalars) {
+  /*--- Compute total source terms from the production and consumption. ---*/
+
+  vector<su2double> lookup_scalar(config->GetNLookups());
+  unsigned long misses = fluid_model_local->EvaluateDataSet(scalars, FLAMELET_LOOKUP_OPS::LOOKUP, lookup_scalar);
+
+  for (auto i_lookup = 0u; i_lookup < config->GetNLookups(); i_lookup++) {
+    nodes->SetLookupScalar(iPoint, lookup_scalar[i_lookup], i_lookup);
+  }
+  return misses;
+}
+
+unsigned long CSpeciesFlameletSolver::GetEnthFromTemp(CFluidModel* fluid_model, su2double const val_temp,
+                                                      su2double* val_enth, const su2double* scalar_solution) {
+  /*--- convergence criterion for temperature in [K], high accuracy needed for restarts. ---*/
+  su2double delta_temp_final = 0.001;
+  su2double enth_iter = scalar_solution[I_ENTH];
+  su2double delta_enth;
+  su2double delta_temp_iter = 1e10;
+  unsigned long exit_code = 0;
+  const int counter_limit = 1000;
+
+  int counter = 0;
+
+  su2double *val_scalars = new su2double[nVar];
```

Comment:
same

Reply:
All news/deletes have been replaced in the `CSpeciesFlameletSolver` class.